design/roadmap.md

```diff
@@ -1,179 +1,71 @@
 
-Currently ksonnet allows you to `ks generate` prototypes from three sources:
-* **[Standard prototypes]** From ksonnet code itself (see [`systemPrototypes.go`](https://github.com/ksonnet/ksonnet/blob/master/prototype/systemPrototypes.go))
-* **[Standard prototypes]** From the [`incubator`](https://github.com/ksonnet/parts/tree/master/incubator) registry in the `ksonnet/parts` Github repo
-* **[Custom prototypes]** From Github directories that conform to the registry spec (see [docs](https://ksonnet.io/docs/cli-reference#ks-registryks-registry-add) for `ks registry add`)
+## Registry
 
-The last option (`ks registry add`) allows users to write and share their own prototypes. *However, this feature is currently limited to Github.* Users may want more flexibility, e.g. a local-filesystem-based registry for development purposes.
+* Users can publish their parts to the Ksonnet registry.
+* Users can offer multiple versions of parts.
 
-**`ks registry add` functionality should be expanded to support other common protocols (local filesystem, S3, etc).** The only requirement thereafter will be having the proper YAML metadata files (e.g. `registry.yaml` in the root).
+## Dependencies
 
-## Easily generate third-party registries
+* Users can configure Ksonnet to use parts on the local filesystem as dependencies.
 
-*Planned for 0.9.0* ([ #234](https://github.com/ksonnet/ksonnet/issues/234))
+## Default parts
 
-Users can create custom, third-party registries, but the `ks registry add` command will only recognize them if they adhere to the registry spec. In other words, they need to have the appropriate YAML files and directory structure.
+* Support for parts based on Kubernetes types will be included with Ksonnet. These parts will act as local dependencies and be upgraded through the registry.
 
-**A new command (e.g. `ks registry create`) which autogenerates registry scaffolding files, may make this easier and less error prone.**
+## Expose environment metadata to prototypes
 
-## Lazily download and update dependencies (`ks install`)
+* Users can reference the current namespace in components.
 
-*Planned for 0.9.0* ([#217](https://github.com/ksonnet/ksonnet/issues/217), [#237](https://github.com/ksonnet/ksonnet/issues/237))
+## JSON and YAML support
 
-Currently, `ks init` fails when the user doesn't have an internet connection, because it relies on two external dependencies:
-* The `k.libsonnet` for the `default` environment
-* The `incubator` registry
+* Users can add JSON, YAML, or Jsonnet file into `components/` and be able to apply them with `ks apply`. Only Jsonnet files can use parameters.
+* Users can use `ks` to convert JSON or YAML files to Jsonnet.
 
-To avoid this issue, **ksonnet needs to lazily initialize environments and registries**, rather than trying to set these up right away.
+## Component references
```

Comment:
I would bump this higher up the list, it seems to be a thing a fair number of people want.